src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs

```diff
@@ -18,13 +22,26 @@ private class ParsedSyntaxTree : CSharpSyntaxTree
             private readonly bool _hasCompilationUnitRoot;
             private readonly Encoding _encodingOpt;
             private readonly SourceHashAlgorithm _checksumAlgorithm;
+            private readonly ImmutableDictionary<string, ReportDiagnostic> _diagnosticOptions;
             private SourceText _lazyText;
 
-            internal ParsedSyntaxTree(SourceText textOpt, Encoding encodingOpt, SourceHashAlgorithm checksumAlgorithm, string path, CSharpParseOptions options, CSharpSyntaxNode root, Syntax.InternalSyntax.DirectiveStack directives, bool cloneRoot = true)
+            internal ParsedSyntaxTree(
```

Comment:
>ParsedSyntaxTree [](start = 21, length = 16)

I would expect to see a way for an external client to create an instance of this class with specific diagnostic options right from the start, but I don't see any.

Reply:
I intentionally didn't provide this because I'm not sure what the most natural API will be. I think that the core code can be tested with the current APIs, and when I add an actual user of the API (e.g., in the command line compiler) I'll add a parsing API and corresponding tests for it.